### DISCUSSION TITLE
What is the difference between MaxTxBatchCount and MaxStateRootElements

### DISCUSSION THREAD

**Discussion by cyborgdennett** [ethereum-optimism/developers#494] [Application Development]
I assume each L2 Tx increments the StateRootElements by one and TxBatch by one.

I assume this is only used for Bedrock, where multiple Tx can be in a single stateroot.

https://github.com/ethereum-optimism/optimism/blob/2e0b633004b14edd14d9c01fd5119b58680a7dfa/batch-submitter/config.go#L85-L91

**Comment by smartcontracts** [comment]
There are two types of batch submitters. They do very similar things, but they submit data to two different locations. The transaction batch submitter posts data to the CanonicalTransactionChain pre-Bedrock and to the special data address post-Bedrock. The state batch submitter posts data to the StateCommitmentChain pre-bedrock and to the L2OutputOracle post-bedrock. 

MaxStateRootElements configures the maximum number of state roots that the state batch submitter will ever try to post in a single transaction. MaxTxBatchCount configures the maximum number of transactions that the transaction batch submitter will ever try to post in a single transaction.

I'm not sure why these are part of the same config object (I haven't looked closely at that part of the codebase).
